browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -172,34 +171,50 @@ BraveBrowserView::BraveBrowserView(std::unique_ptr<Browser> browser)
                           base::Unretained(this)));
 #endif
 
-#if BUILDFLAG(ENABLE_SIDEBAR)
+  const bool can_have_vertical_tabs =
+      browser_->is_type_normal() && tabs::features::ShouldShowVerticalTabs();
+
   // Only normal window (tabbed) should have sidebar.
-  if (!sidebar::CanUseSidebar(browser_.get())) {
+  const bool can_have_sidebar = sidebar::CanUseSidebar(browser_.get());
+
+  const bool need_to_wrap_contents_container =
+      can_have_vertical_tabs || can_have_sidebar;
+  if (!need_to_wrap_contents_container)
     return;
-  }
 
   auto brave_contents_container = std::make_unique<views::View>();
 
   // Wrap |contents_container_| within our new |brave_contents_container_|.
   // |brave_contents_container_| also contains sidebar.
   auto orignal_contents_container = RemoveChildViewT(contents_container_.get());
-  sidebar_container_view_ = brave_contents_container->AddChildView(
-      std::make_unique<SidebarContainerView>(GetBraveBrowser()));
+
+  if (can_have_sidebar) {
+    sidebar_container_view_ = brave_contents_container->AddChildView(
+        std::make_unique<SidebarContainerView>(GetBraveBrowser()));
+  }
+
+  if (can_have_vertical_tabs) {
+    vertical_tabs_container_ = brave_contents_container->AddChildView(
+        std::make_unique<VerticalTabStripContainer>(tab_strip_region_view_));
+  }
+
   original_contents_container_ = brave_contents_container->AddChildView(
       std::move(orignal_contents_container));
+
   brave_contents_container->SetLayoutManager(
       std::make_unique<BraveContentsLayoutManager>(
-          sidebar_container_view_, original_contents_container_));
+          sidebar_container_view_, vertical_tabs_container_,
```

Comment:
sidebar or vertical tabs might be null here because one does not imply the other